test/com/google/enterprise/adaptor/database/UniqueKeyTest.java

```diff
@@ -104,6 +104,32 @@ public void testNameRepeatsNotAllowed() {
   }
 
   @Test
+  public void testCaseSensitiveNameRepeatsAllowed() {
+    UniqueKey uk = new UniqueKey("num:int,Num:string");
+    assertEquals(2, uk.numElementsForTest());
+    assertEquals("num", uk.nameForTest(0));
+    assertEquals("Num", uk.nameForTest(1));
+    assertEquals(UniqueKey.ColumnType.INT, uk.typeForTest(0));
+    assertEquals(UniqueKey.ColumnType.STRING, uk.typeForTest(1));
+  }
+
+  @Test
+  public void testCaseInsensitiveNameNotAllowedIfCaseSensitiveKeys() {
+    thrown.expect(InvalidConfigurationException.class);
+    UniqueKey uk = new UniqueKey("id:int,Id:string", "ID", "ID");
```

Comment:
I mentioned this week that I had explicitly rejected this logic. If the database has case-sensitive column names, I don't think we can predict whether a lookup with yet another case will succeed or not.

I haven't thought this completely through, but I think if you make the tmpTypes keys case-insensitive, with the String.CASE_INSENSITIVE_ORDER, you might not have to do anything else.